sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -48,3 +48,38 @@ case class Md5(child: Expression)
         s"(org.apache.commons.codec.digest.DigestUtils.md5Hex($c))")
   }
 }
+
+/**
+ * A function that calculates a sha1 hash value and returns it as a hex string
+ * For input of type [[BinaryType]] or [[StringType]]
+ */
+case class Sha1(child: Expression) extends UnaryExpression {
+
+  override def dataType: DataType = StringType
```

Comment:
What's the expected type of child? Which could help to provide better error messages